src/api/java/com/minecolonies/api/entity/citizen/happiness/TimeBasedHappinessModifier.java

```diff
@@ -0,0 +1,95 @@
+package com.minecolonies.api.entity.citizen.happiness;
+
+import com.minecolonies.api.util.Tuple;
+import net.minecraft.nbt.CompoundNBT;
+
+import java.util.function.DoubleSupplier;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_DAY;
+
+/**
+ * The time based happiness modifier.
+ */
+public class TimeBasedHappinessModifier extends StaticHappinessModifier
+{
+    /**
+     * The time based factors.
+     */
+    private final Tuple<Integer, Double>[] timeBasedFactor;
+
+    /**
+     * The number of passed days.
+     */
+    private int days = 0;
+
+    /**
+     * Create an instance of the happiness modifier.
+     *
+     * @param id     its string id.
+     * @param weight its weight.
```

Comment:
missing javadoc, as supplier and timebasedfactor are both undocumented and actually important parts for understanding this